### PR TITLE
Remove source/message.h

### PR DESCRIPTION
**Pull request by dj2**
The code in source/message was only used in a single set of tests to
format the output results. This CL changes the test to verify the
message instead of all the error values and removes the source/message
code.

### CODE REVIEW

File: test/opt/ir_loader_test.cpp [hunk 1]
```diff
@@ -322,10 +321,9 @@ TEST(IrBuilder, KeepModuleProcessedInRightPlace) {
 // from the given |assembly|.
 void DoErrorMessageCheck(const std::string& assembly,
                          const std::string& error_message) {
-  auto consumer = [error_message](spv_message_level_t level, const char* source,
-                                  const spv_position_t& position,
-                                  const char* m) {
-    EXPECT_EQ(error_message, StringifyMessage(level, source, position, m));
+  auto consumer = [error_message](spv_message_level_t, const char*,
+                                  const spv_position_t&, const char* m) {
+    EXPECT_EQ(error_message, m);
```
Comment (s-perron) [inline test/opt/ir_loader_test.cpp]: This change no longer checks that the error was found in the correct position.  It that really what we want?

Comment (s-perron) [reply test/opt/ir_loader_test.cpp]: That could be why we ended up with such bad messages coming out of the validator.  I really think we should be checking the position as well.  I don't want these tests to check for less than they currently do.

Comment (dj2) [reply test/opt/ir_loader_test.cpp]: Done.

**Comment by dj2** [comment]
Changed to verify message and line number.

**Comment by johnkslang** [comment]
Note that deletion of files breaks down-stream users.

**Comment by johnkslang** [comment]
> Note that deletion of files breaks down-stream users.

Apparently that was improperly consumed, sorry for the distraction.

**Pull request merged**